village/src/App.js

```diff
@@ -1,27 +1,68 @@
-import React, { Component } from 'react';
+import React, { useState, useEffect } from 'react';
+import axios from 'axios';
+import { Route, NavLink } from 'react-router-dom';
 
 import './App.css';
 import SmurfForm from './components/SmurfForm';
 import Smurfs from './components/Smurfs';
 
-class App extends Component {
-  constructor(props) {
-    super(props);
-    this.state = {
-      smurfs: [],
-    };
-  }
+const App = () => {
+  const [smurfs, setSmurfs] = useState([]);
+
   // add any needed code to ensure that the smurfs collection exists on state and it has data coming from the server
   // Notice what your map function is looping over and returning inside of Smurfs.
   // You'll need to make sure you have the right properties on state and pass them down to props.
-  render() {
-    return (
-      <div className="App">
-        <SmurfForm />
-        <Smurfs smurfs={this.state.smurfs} />
-      </div>
-    );
+
+  const [mounted, setMounted] = useState(false);
+  useEffect(() => {
+    !mounted &&
+      axios
+        .get('http://localhost:3333/smurfs')
+        .then(res => setSmurfs(res.data))
+        .catch(err => console.log(err))
+    setMounted(true);
+  })
```

Comment:
To use `useEffect` for a ComponentDidMount life cycle, pass an empty array as a second argument to the invocation. 
```
useEffect(() => {
  axios
        .get('http://localhost:3333/smurfs')
        .then(res => setSmurfs(res.data))
        .catch(err => console.log(err))
}, [])
```

This way, you don't need to use `mounted` state at all. Plus, using `setMounted` will always fire the useEffect every time the component updated, but adding the empty array will make sure that the useEffect will only be called once after the component first render.